addons/website/static/src/js/website.contentMenu.js

```diff
@@ -70,6 +70,40 @@
                 }
             });
         },
+        rename_page: function() {
+            var self = this;
+            var context = website.get_context();
+            self.mo_id = self.getMainObject().id;
+
+            openerp.jsonRpc('/web/dataset/call_kw', 'call', {
+                model: 'website',
+                method: 'page_search_dependencies',
+                args: [self.mo_id],
+                kwargs: {
+                    context: context
+                },
+            }).then(function (deps) {
+                website.prompt({
+                    id: "editor_rename_page",
+                    window_title: _t("Rename Page"),
+                    dependencies: deps,
+                }, 'website.rename_page').then(function (val, field, $dialog) {
+                    openerp.jsonRpc('/web/dataset/call_kw', 'call', {
+                        model: 'website',
+                        method: 'rename_page',
+                        args: [
+                            self.mo_id,
+                            val,
+                        ],
+                        kwargs: {
+                            context: context
+                        },
+                    }).then(function () {
+                        window.location = "/page/" + encodeURIComponent(val);
```

Comment:
probably need to use the version returned by the controller, after `slugify` call